### PR TITLE
Feature/self hosted f4 e sync

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Adds a random, daily job to the API project for a self hosted sponsorship sync. 

Some of these changes were left over from a strategy of adding one trigger per organization. I don't see the harm in leaving them in, as they allow us to add more complicated jobs in the future.

## Code changes

* **src/Core/Settings/GlobalSettings.cs:** add `EnableCloudCommunication` setting
* **src/Api/Jobs/SelfHostedSponsorshipSyncJob.cs:** add job for syncing sponsorships
* **src/Core/Jobs/BaseJobsHostedService.cs:** set Jobs array to take `IJobDetail` instead of just a type. We can then create different jobs in the future inside each projects implementation.
* **src/Api/Jobs/JobsHostedService.cs:** create random daily trigger and self hosted sync job, adding them to the scheduler
* **src/Api/Startup.cs:** pass in global settings to job scheduler and remove extra code

## Before you submit
- [ ] I have checked for formatting errors (`dotnet tool run dotnet-format --check`) (required)
- [ ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: src/Core/Settings/GlobalSettings.cs [hunk 1]
```diff
@@ -37,6 +37,7 @@ public virtual string LicenseDirectory
         public virtual string HibpApiKey { get; set; }
         public virtual bool DisableUserRegistration { get; set; }
         public virtual bool DisableEmailNewDevice { get; set; }
+        public virtual bool EnableCloudCommunication { get; set; }
```
Comment (MGibson1) [inline src/Core/Settings/GlobalSettings.cs:40]: We want to make sure this defaults to false.

**Review by MGibson1 — CHANGES_REQUESTED** [review]
The scaffolding for the new job looks good, We need to get Justin's API work in so we can pull Orgs with billing sync keys to iterate through.

File: src/Api/Jobs/JobsHostedService.cs [hunk 2]
```diff
@@ -46,21 +46,38 @@ public override async Task StartAsync(CancellationToken cancellationToken)
...
+            var randomDailySponsorshipSyncTrigger = TriggerBuilder.Create()
+                .WithIdentity("randomDailySponsorshipSyncTrigger")
+                .StartAt(DateBuilder.FutureDate(new Random().Next(24), IntervalUnit.Hour)) 
+                .WithSimpleSchedule(x => x
+                    .WithIntervalInHours(24)
+                    .RepeatForever())
+                .Build();
 
-            Jobs = new List<Tuple<Type, ITrigger>>
+            var jobs = new List<Tuple<IJobDetail, ITrigger>>
```
[the remark this answers is not in the document]
Comment (jlf0dev) [reply src/Api/Jobs/JobsHostedService.cs:57]: The `JobDetail.Key` looks like `DEFAULT.Bit.Api.Jobs.AliveJob`, meaning we won't lose out on any logging quality.

If the dangers outweigh the benefits of this change I can revert, but with our original implementation we could only schedule one type of job.

**Comment by jlf0dev** [comment]
moved to #1922

**Pull request closed without merging**